pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -189,11 +200,45 @@ func findImportOvaParams(client *vim25.Client, datacenter, cluster, datastore, n
 	for _, hostObj := range hosts {
 		foundDatastore := false
 		foundNetwork := false
-		err := hostObj.Properties(ctx, hostObj.Reference(), []string{"network", "datastore", "runtime"}, &hostSystemManagedObject)
+		err := hostObj.Properties(ctx, hostObj.Reference(), []string{"config.product", "network", "datastore", "runtime"}, &hostSystemManagedObject)
 		if err != nil {
 			return nil, err
 		}
 
+		// If HardwareVersion is 13 there is no reason to continue checking
+		// There is a ESXi host that does not support hardware 15.
+		if importOvaParams.HardwareVersion != "vmx-13" {
+			esxiHostVersion, err := version.NewVersion(hostSystemManagedObject.Config.Product.Version)
+			if err != nil {
+				return nil, err
+			}
+			// Check if ESXi host is greater than or equal to 6.7
+			if constraints.Check(esxiHostVersion) {
```

Comment:
The logic that I have here for constraints needs to be refined. Wanted to get a PR open to test with CI